src/blockdata/transaction.rs

```diff
@@ -437,11 +438,11 @@ impl Transaction {
     }
 }
 
-impl BitcoinHash for Transaction {
-    fn bitcoin_hash(&self) -> sha256d::Hash {
-        let mut enc = sha256d::Hash::engine();
+impl BitcoinHash<Txid> for Transaction {
+    fn bitcoin_hash(&self) -> Txid {
```

Comment:
Yeah this is not correct. The txid skips the witness data. This should be `Wtxid`. I'm not sure I'm a fan of the `BitcoinHash` trait in this context.

Reply:
And I assume there is no test coverage for Wtxid since it was left unattended.

Should I fix this and add tests in a separate PR or its better to do it here?

Reply:
Fix it for sure. Adding a unit test shouldn't be hard. Thanks!

Reply:
Done!